pkg/vsphere/vm/vm.go

```diff
@@ -121,6 +121,25 @@ func (vm *VirtualMachine) FolderName(ctx context.Context) (string, error) {
 	return path.Base(u.Path), nil
 }
 
+// Folder returns a reference to the parent folder that owns the vm
+func (vm *VirtualMachine) Folder(ctx context.Context) (*object.Folder, error) {
+	// NOTE: We must retrieve the folder by assembling the path. If the vm is in a vApp the inventory path will be the compute folder path of the vApp.
+	// right now this will likely not work with upgrade 1.1.1-1.2.1 vapp based deployments.
```

Comment:
Do we have a TODO or a strategy to fix this in the future?

Reply:
Not at the moment. Left the NOTE: here since we are working to deprecate the vApp deployment method. Since this is the case this code should work well if we are not going to attempt an upgrade. After talking with @cgtexmex this will likely be a problem. we talked some about it yesterday. @cgtexmex thoughts? This section will be affected by the vApp potentially and without an upgrade right now it is hard to test at the moment.